WalletWasabi/CoinJoin/Coordinator/Rounds/CoordinatorRound.cs

```diff
@@ -1177,35 +1177,37 @@ public async Task<IEnumerable<Alice>> RemoveAlicesIfAnInputRefusedByMempoolAsync
 					throw new InvalidOperationException("Removing Alice is only allowed in InputRegistration and ConnectionConfirmation phases.");
 				}
 
-				var checkingTasks = new List<(Alice alice, Task<(bool accepted, string reason)> task)>();
-				var batch = RpcClient.PrepareBatch();
-				foreach (Alice alice in Alices)
+				// Check if mempool would accept a fake transaction created with all the registered inputs.
+				var coinsToTest = Alices.SelectMany(alice => alice.Inputs);
+				// Add the outputs by denomination level.
+				var outputsAll = Alices.Sum(alice => EstimateBestMixingLevel(alice));
+				// Add the change outputs.
+				outputsAll += Alices.Count;
+
+				var resultAll = await RpcClient.TestMempoolAcceptAsync(coinsToTest, fakeOutputsCount: outputsAll);
+				if (resultAll.accept)
 				{
-					// Check if mempool would accept a fake transaction created with the registered inputs.
-					// This will catch ascendant/descendant count and size limits for example.
-					checkingTasks.Add((alice, batch.TestMempoolAcceptAsync(alice.Inputs)));
+					return Enumerable.Empty<Alice>();
 				}
-				var waiting = Task.WhenAll(checkingTasks.Select(t => t.task));
-				await batch.SendBatchAsync().ConfigureAwait(false);
-				await waiting.ConfigureAwait(false);
+				Logger.LogInfo($"Mempool acceptance is unsuccessful! Number of Alices: {Alices.Count}.");
 
-				foreach (var t in checkingTasks)
+				// The created tx was not accepted. Let's only accept confirmed txs then.
+				foreach (var alice in Alices.Where(x => !x.AllInputsConfirmed))
 				{
-					var result = await t.task.ConfigureAwait(false);
-					if (!result.accepted)
+					foreach (var input in alice.Inputs.Select(x => x.Outpoint))
 					{
-						alicesRemoved.Add(t.alice);
-						Alices.Remove(t.alice);
-						Logger.LogInfo($"Mempool acceptance failed: {result.reason ?? "no reason"}.");
+						var response = await RpcClient.GetTxOutAsync(input.Hash, (int)input.N, includeMempool: true);
```

Comment:
No more batching rpc request then? Is this intentional?

Reply:
The `gettxout` rpc returns the `unspent` output. In case the it is spent but the spending transaction is not confirmed yet the `gettxout` will not return the uxto neither, in this case it will fail with an exception and `Alice` should be removed because she spent the utxo after registering it with the coordinator.

Reply:
I was worng, it doesn't throw exception but returns `null` instead so, we have to check it here and remove Alice.

Reply:
I miss the batching too, @nopara73 why did you remove it?

Do not forget ConfigureAwait(false).

Reply:
I like the idea. If we got any mempool rejection we fall back where we allow only confirmed inputs.

Reply:
I was too tired to implement batching.

Reply:
@lontivero that's a good point. Previously this code was a double spend protection too, now it isn't. I'm rolling it back to check all inputs, not only the unconfirmed ones.

Reply:
Regarding batching, I think that's dangerous now, but may be justified. Previously we knew that the maximum requests we gave out was 100 (since 100 peers.) Which works fine with the RPC parallelization. This time the max requests are 700, as 100 peers * 7 inputs, which does not work fine with the RPC parallelization. Sure more complex algorithms can be implemented though, but is it necessary? (It's not a rhetoric question, I genuinely don't know.)